apps/src/netsim/NetSimHeartbeat.js

```diff
@@ -133,12 +141,30 @@ NetSimHeartbeat.prototype.setBeatInterval = function (intervalMs) {
 };
 
 /**
+ * Set a handler to call if this heartbeat is unable to update its remote
+ * storage representation.  Can be used to go into a recovery mode,
+ * acknowledge disconnect, and/or attempt an auto-reconnect.
+ * @param {function} onFailedHeartbeat
+ */
+NetSimHeartbeat.prototype.setFailureCallback = function (onFailedHeartbeat) {
+  this.onFailedHeartbeat_ = onFailedHeartbeat;
```

Comment:
Should we insert a check that a failedHeartbeat_ callback hasnt already been added?